test/E2ETest/TyePurgeTests.cs

```diff
@@ -0,0 +1,159 @@
+
+        public TyePurgeTests(ITestOutputHelper output)
+        {
+            this.output = output;
+            sink = new TestOutputLogEventSink(output);
+        }
+
+        [Fact]
+        public async Task FrontendBackendPurgeTest()
+        {
+            var projectDirectory = new DirectoryInfo(Path.Combine(TestHelpers.GetSolutionRootDirectory("tye"), "samples", "frontend-backend"));
+            var tempDirectory = TempDirectory.Create();
+            DirectoryCopy.Copy(projectDirectory.FullName, tempDirectory.DirectoryPath);
+
+            var projectFile = new FileInfo(Path.Combine(tempDirectory.DirectoryPath, "tye.yaml"));
+            var tyeDir = new DirectoryInfo(Path.Combine(tempDirectory.DirectoryPath, ".tye"));
+            using var host = new TyeHost(ConfigFactory.FromFile(projectFile).ToHostingApplication(), Array.Empty<string>())
+            {
+                Sink = sink,
+            };
+
+            await host.StartAsync();
+            try
+            {
+                await WaitUntilSpawned(host.Application);
+
+                var pids = GetAllPids(host.Application);
+
+                Assert.True(Directory.Exists(tyeDir.FullName));
+                Assert.True(AllRunning(pids));
+
+                await host.PurgeAsync();
+                await Task.Delay(500);
+
+                Assert.False(AnyRunning(pids));
+            }
+            finally
+            {
+                await host.StopAsync();
+            }
+
+            try
+            {
+                tempDirectory.Dispose();
+            }
+            catch (UnauthorizedAccessException) { }
+        }
+
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task MultiProjectPurgeTest()
+        {
+            var projectDirectory = new DirectoryInfo(Path.Combine(TestHelpers.GetSolutionRootDirectory("tye"), "samples", "multi-project"));
+            var tempDirectory = TempDirectory.Create();
+            DirectoryCopy.Copy(projectDirectory.FullName, tempDirectory.DirectoryPath);
+
+            var projectFile = new FileInfo(Path.Combine(tempDirectory.DirectoryPath, "tye.yaml"));
+            var tyeDir = new DirectoryInfo(Path.Combine(tempDirectory.DirectoryPath, ".tye"));
+            using var host = new TyeHost(ConfigFactory.FromFile(projectFile).ToHostingApplication(), Array.Empty<string>())
+            {
+                Sink = sink,
+            };
+
+            await host.StartAsync();
+            try
+            {
+                await WaitUntilSpawned(host.Application);
+
+                var pids = GetAllPids(host.Application);
+                var containers = GetAllContainerIds(host.Application);
+
+                Assert.True(Directory.Exists(tyeDir.FullName));
+                Assert.True(AllRunning(pids));
+                await DockerAssert.AssertAllContainersExistAsync(output, containers);
+
+                await host.PurgeAsync();
+                await Task.Delay(500);
```

Comment:
What's the reason for the wait? Generally if there are cases where you need to put in a wait, we'll eventually find a CI machine that is too slow 😆

Reply:
yeah. I wanted to avoid it but it seems that the kill of a process/container is asynchronous so without the delay the test would sometimes pass but sometimes won't
I can probably do the same thing that I've done with `WaitUntilSpawned` (e.g. `WaitUntilDespawned`)

Reply:
The tests also fail for other reasons in rare cases. For example, `WaitUntilSpawned` waits until the replica count equals to the replica count in the definition (for all services), but it doesn't wait until all replicas are of `Running` state, because the state is not saved in the ReplicaStatus class.
Maybe I should rethink my testing strategy. Maybe go with what was done in the other test classes (creating an HttpClient that interacts with the services directly)